feature (api): create simple fastapi server

bukkalexander opened the pull request:
Created a FastAPI server and connected the frontend to it locally.

bukkalexander committed:
feature (api): create simple fastapi server and connect it to the frontend

mghnm reviewed (commented):
The use of fast_api to deploy a webserver might not be appropriate when using api_gateway and lambda. This is for the reason that api_gateway in this case acts as a proxy for lambda. Every time it receives a request it will invoke lambda.

bukkalexander commented:
You are right, but there is a workaround using a library called 'mangum' that will make it work inside a lambda automatically, according to chatgpt at least 😅 But we can use plain functions also, whatever feels best

bukkalexander commented:
The only reason I'd like to use fastapi is that I wanted to learn how to use it, and it will be simple (I think) to use it both locally when developing and remotely when deployed as lambda, since the routing can be tested locally, and when deployed, we can configure it to use mangum to handle the event and convert it to the fastapi interface. Also, if we decide to switch to ec2, we already get an easy api server setup for free.

Example

```

from fastapi import FastAPI
from mangum import Mangum

app = FastAPI()

@app.get("/")
def read_root():
    return {"message": "Hello from Lambda!"}

# Add Mangum handler to work with AWS Lambda
handler = Mangum(app)

````

mghnm committed:
Merge master into pr

mghnm committed:
Add make targets for FE build and expand infra to hold it

bukkalexander committed:
feature (repo): add mangum

Also, add possibility to use VITE_API_URL environemnt variable
to confugure question fetching url during the deployment,
instead of host and port for local testing

bukkalexander committed:
feature (web): fetch api_url from config.json

mghnm committed:
make sure we build the frontend as part of the flow

mghnm committed:
Merge branch 'feature/fastapi' of github.com:bukkalexander/corplife into feature/fastapi

mghnm committed:
path to code wrong sinze zip was archivnig full path

mghnm committed:
fix: wrong contents in archive

bukkalexander committed:
fix (web): update config names

bukkalexander committed:
refactor (api): separate local and production entrypoints

mghnm committed:
fix: use mangum class not model

mghnm commented:
Application main scenario is in working state. FE is communicating with BE and fetching the test questions from the APP. Next step would be throwing those questions into the DynamoDB table for complete integration.

ifayazagreat commented on web/package.json:
```diff
@@ -14,6 +14,7 @@
     "@emotion/react": "^11.13.3",
     "@emotion/styled": "^11.13.0",
     "@mui/material": "^6.1.4",
+    "dotenv": "^16.4.5",
```
remove

ifayazagreat reviewed (commented):
see review comment

bukkalexander committed:
fix (web): uninstall dotenv
